addons/gamification/models/gamification_badge.py

```diff
@@ -0,0 +1,229 @@
+
+    stat_count = fields.Integer(
+        compute='_compute_owners_info', string='Total',
+        help="The number of time this badge has been received.")
+    stat_count_distinct = fields.Integer(
+        compute='_compute_owners_info',
+        string='Number of users', help="The number of time this badge has been received by unique users.")
+    stat_this_month = fields.Integer(
+        compute='_compute_badge_user_stats',
+        string='Monthly total', help="The number of time this badge has been received this month.")
+    stat_my = fields.Integer(
+        compute='_compute_badge_user_stats', string='My Total',
+        help="The number of time the current user has received this badge.")
+    stat_my_this_month = fields.Integer(
+        compute='_compute_badge_user_stats',
+        string='My Monthly Total', help="The number of time the current user has received this badge this month.")
+
+    def _compute_badge_user_stats(self):
+        """Return stats related to badge users"""
+        BaseUser = self.env['gamification.badge.user']
+        today = fields.Date.from_string(fields.Date.today())
+        first_month_day = fields.Date.to_string(today.replace(day=1))
+        for bid in self:
+            bid.stat_my = BaseUser.search_count([('badge_id', '=', bid.id), ('user_id', '=', self.env.uid)])
+            bid.stat_this_month = BaseUser.search_count([('badge_id', '=', bid.id), ('create_date', '>=', first_month_day)])
+            bid.stat_my_this_month = BaseUser.search_count([('badge_id', '=', bid.id), ('user_id', '=', self.env.uid), ('create_date', '>=', first_month_day)])
+            bid.stat_my_monthly_sending = BaseUser.search_count([('badge_id', '=', bid.id), ('create_uid', '=', self.env.uid), ('create_date', '>=', first_month_day)])
+
+    def _compute_remaining_sending(self):
+        """Computes the number of badges remaining the user can send
+
+        0 if not allowed or no remaining
+        integer if limited sending
+        -1 if infinite (should not be displayed)
+        """
+        for badge in self:
+            if badge._can_grant_badge() != 1:
+                # if the user cannot grant this badge at all, result is 0
+                badge.remaining_sending = 0
+            elif not badge.rule_max:
+                # if there is no limitation, -1 is returned which means 'infinite'
+                badge.remaining_sending = -1
+            else:
+                badge.remaining_sending = badge.rule_max_number - badge.stat_my_monthly_sending
+
+    def _compute_owners_info(self):
+        """Return:
+            the list of unique res.users ids having received this badge
+            the total number of time this badge was granted
+            the total number of users this badge was granted to
+        """
+        for bid in self:
+            bid.stat_count = 0
+            bid.stat_count_distinct = 0
+            bid.unique_owner_ids = []
```

Comment:
Note that if we do not write default value then we will get falsy value, and list view(for e.g.) will display nothing in cell(If it is acceptable then it is not required)